metrics/sliding_time_window_sample.go

```
@@ -0,0 +1,191 @@
+package metrics
+
+// go version of sliding window reservoir
+// refer to : https://github.com/dropwizard/metrics/blob/3.2-development/metrics-core/src/main/java/com/codahale/metrics/SlidingTimeWindowReservoir.java
+// and : https://github.com/rcrowley/go-metrics/blob/master/sample.go
+import (
+	"sync"
+	"time"
+
+	gomet "github.com/rcrowley/go-metrics"
+)
+
+const (
+	collisionBuffer int64 = 256
+	trimThreshold   int64 = 256
+	clearBuffer     int64 = int64(time.Hour/time.Nanosecond) * 256
+)
+
+// SlidingTimeWindowSample : http://metrics.dropwizard.io/3.1.0/manual/core/#sliding--time-window-reservoirs
+type SlidingTimeWindowSample struct {
+	window   int64
+	lastTick int64
+	count    int64
+	mutex    sync.Mutex
+	values   map[int64]int64
+}
+
+// NewSlidingTimeWindowSample : New
+func NewSlidingTimeWindowSample(duration string) gomet.Sample {
+	if gomet.UseNilMetrics {
+		return gomet.NilSample{}
+	}
+	sample := &SlidingTimeWindowSample{
+		values:   make(map[int64]int64),
+		lastTick: time.Now().UnixNano() * collisionBuffer,
+	}
+	if dur, err := time.ParseDuration(duration); err == nil {
+		sample.window = dur.Nanoseconds() * collisionBuffer
```

We shouldn't just hide the error, if there is one, then return the error.

---

also you should open a pr back to go metrics with this change